x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/Connector.java

```diff
@@ -0,0 +1,753 @@
+        PARSER.declareBoolean(optionalConstructorArg(), IS_NATIVE_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LANGUAGE_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LAST_ACCESS_CONTROL_SYNC_ERROR);
+        PARSER.declareField(
+            optionalConstructorArg(),
+            (p, c) -> ConnectorSyncStatus.connectorSyncStatus(p.text()),
+            LAST_ACCESS_CONTROL_SYNC_STATUS_FIELD,
+            ObjectParser.ValueType.STRING
+        );
+        PARSER.declareString(optionalConstructorArg(), LAST_ACCESS_CONTROL_SYNC_SCHEDULED_AT_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LAST_DELETED_DOCUMENT_COUNT_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LAST_INCREMENTAL_SYNC_SCHEDULED_AT_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LAST_INDEXED_DOCUMENT_COUNT_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LAST_SEEN_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LAST_SYNC_ERROR_FIELD);
+        PARSER.declareString(optionalConstructorArg(), LAST_SYNC_SCHEDULED_AT_FIELD);
+        PARSER.declareField(
+            optionalConstructorArg(),
+            (p, c) -> ConnectorSyncStatus.connectorSyncStatus(p.text()),
+            LAST_SYNC_STATUS_FIELD,
+            ObjectParser.ValueType.STRING
+        );
+        PARSER.declareString(optionalConstructorArg(), LAST_SYNCED_FIELD);
+        PARSER.declareString(optionalConstructorArg(), NAME_FIELD);
+        PARSER.declareField(
+            optionalConstructorArg(),
+            (p, c) -> ConnectorIngestPipeline.fromXContent(p),
+            PIPELINE_FIELD,
+            ObjectParser.ValueType.OBJECT
+        );
+        PARSER.declareField(
+            optionalConstructorArg(),
+            (p, c) -> ConnectorScheduling.fromXContent(p),
+            SCHEDULING_FIELD,
+            ObjectParser.ValueType.OBJECT
+        );
+        PARSER.declareString(optionalConstructorArg(), SERVICE_TYPE_FIELD);
+        PARSER.declareField(
+            optionalConstructorArg(),
+            (p, c) -> ConnectorStatus.connectorStatus(p.text()),
+            STATUS_FIELD,
+            ObjectParser.ValueType.STRING
+        );
+        PARSER.declareField(
+            optionalConstructorArg(),
+            (parser, context) -> parser.map(),
+            SYNC_CURSOR_FIELD,
+            ObjectParser.ValueType.OBJECT_OR_NULL
+        );
+        PARSER.declareString(optionalConstructorArg(), SYNC_NOW_FIELD);
+    }
+
+    public static Connector fromXContentBytes(String connectorId, BytesReference source, XContentType xContentType) {
+        try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, source, xContentType)) {
+            return Connector.fromXContent(connectorId, parser);
+        } catch (IOException e) {
+            throw new ElasticsearchParseException("Failed to parse: " + source.utf8ToString(), e);
```

Comment:
Are there PII concerns logging this source? I see things like API keys so we might want to be careful here.

Reply:
That's correct, good spot! The configuration might contain client secrets (api keys to external sources). I will limit the amount of information we log.

Reply:
I realised that I don't need parser for the `Connector` class - as it deals only with internal representation of the data in the index. 

I will keep your comment in mind in subsequent PRs that deal with passing/updating PIIs :+1: